lib/orbit/transformable.js

```diff
@@ -1,121 +1,118 @@
 import Orbit from './main';
 import Evented from './evented';
 import ActionQueue from './action-queue';
-import Transformation from './transformation';
-import Operation from './operation';
-import { isArray } from './lib/objects';
+import Transform from './transform';
+import TransformResult from './transform-result';
 import { assert } from './lib/assert';
+import { deprecate } from './lib/deprecate';
 
-function normalize(operation) {
-  if (isArray(operation)) {
-    return operation.map(function(o) {
-      return normalize(o);
-    });
-
-  } else {
-    if (operation instanceof Operation) {
-      return operation;
-    } else {
-      return new Operation(operation);
-    }
-  }
+function settleTransform(transform, result) {
+  var _this = this;
+
+  this._transformLog[transform.id] = true;
+
+  return this.settle.call(this, 'didTransform', transform, result);
 }
 
-function transformationFor(operation) {
-  var transformation;
-  var i;
-
-  if (isArray(operation)) {
-    for (i = 0; i < operation.length; i++) {
-      var t = transformationFor.call(this, operation[i]);
-      if (transformation) {
-        if (t !== transformation) return;
-      } else {
-        transformation = t;
-      }
-    }
-    return transformation;
+function processTransform(transform) {
+  // console.log('processTransform', this.id, transform);
 
-  } else {
-    var queue = this._transformationQueue.content;
+  var _this = this;
+  var ops = transform.operations;
+  var result;
+
+  if (this.prepareTransformOperations) {
+    ops = this.prepareTransformOperations(ops);
+  }
 
-    // console.log('transformationFor', operation, queue.length);
+  if (ops.length > 0) {
+    var response = this._transform(ops);
 
-    for (i = 0; i < queue.length; i++) {
-      transformation = queue[i].data;
-      if (transformation.verifyOperation(operation)) {
-        return transformation;
+    if (response) {
+      if (response.then) {
+        return response.then(function(result) {
+          return settleTransform.call(_this, transform, result);
+        });
+      } else if (response instanceof TransformResult) {
+        result = response;
       }
     }
   }
+
+  if (!result) result = new TransformResult();
+
+  return settleTransform.call(this, transform, result);
+}
+
+function queueTransform(transform) {
+  var _this = this;
+
+  var action = this._transformQueue.push({
+    data: transform,
+    process: function() {
+      return processTransform.call(_this, transform);
+    }
+  });
+
+  return action.complete;
 }
 
-function queueTransformation(transformation) {
+function queueTransformResult(transform, result) {
   var _this = this;
 
-  var processor = this._transformationQueue.push({
-    data: transformation,
+  var action = this._transformQueue.push({
```

Comment:
In this scenario the transform has already completed, shouldn't it "jump the queue" so to speak? We're perhaps hitting the limits of a queue based on promises, it's append only :)